os-audit/cos-auditd-logging.yaml

```
@@ -71,7 +71,7 @@ spec:
           value: "$CLUSTER_NAME"
         - name: CLUSTER_LOCATION
           value: "$CLUSTER_LOCATION"
-        image: gke.gcr.io/fluent-bit@sha256:b148f7f960f101b6d52efd909fe43fef73cb40cee3571da61034974965605b66 # v1.8.12-gke.31
+        image: fluent/fluent-bit:latest
```

is it a good practice to use latest and suggest to use one from dockerhub?

---

I discussed internally with my teammates and we believe there is no harm for the cx to use OSS version of image in this particular case since we only provide an example to setup audit logging in COS.

Regarding using `latest`, the purpose is to reduce potential cx tickets due to mismatched version of `libsystemd` used to compile fluent-bit and the one used in COS. We agree that it will break at some point (e.g fluent-bit change config format etc) but it is fine to update this file again later.

---

We discussed internally and we support this solution in a "best effort" way and would like to highlight that this is a solution depending on OSS tool - hence suggesting use the latest dockerhub image.

---

Please add a comment above this line suggesting to change it. Something like:

```suggestion
        # this image is used for demo purposes. The best practice is to use the image from controlled registry and reference it by SHA.
        image: fluent/fluent-bit:latest
```

---

Extra credit for mentioning this in the docs: https://cloud.google.com/kubernetes-engine/docs/how-to/linux-auditd-logging as well

---

The notes have been added and we will update the public doc later.